src/components/application_manager/src/commands/button_notification_to_mobile.cc

```diff
@@ -0,0 +1,168 @@
+#include "application_manager/commands/button_notification_to_mobile.h"
+namespace application_manager {
+
+namespace commands {
+namespace app_mngr = application_manager;
+
+CREATE_LOGGERPTR_GLOBAL(logger_, "ButtonNotificationToMobile")
+
+ButtonNotificationToMobile::ButtonNotificationToMobile(
+    const app_mngr::commands::MessageSharedPtr& message,
+    app_mngr::ApplicationManager& application_manager,
+    app_mngr::rpc_service::RPCService& rpc_service,
+    app_mngr::HMICapabilities& hmi_capabilities,
+    policy::PolicyHandlerInterface& policy_handler)
+    : CommandNotificationImpl(message,
+                              application_manager,
+                              rpc_service,
+                              hmi_capabilities,
+                              policy_handler) {}
+
+ButtonNotificationToMobile::~ButtonNotificationToMobile() {}
+
+bool ButtonNotificationToMobile::DoesAppIDExist() const {
+  LOG4CXX_AUTO_TRACE(logger_);
+  using namespace application_manager;
+  return (*message_)[strings::msg_params].keyExists(strings::app_id);
+}
+
+void ButtonNotificationToMobile::HandleCustomButton(
+    app_mngr::ApplicationSharedPtr app) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  using namespace application_manager;
+  // app_id is mandatory for CUSTOM_BUTTON notification
+  if (!DoesAppIDExist()) {
+    LOG4CXX_ERROR(logger_, "CUSTOM_BUTTON mobile notification without app_id.");
+    return;
+  }
+
+  // custom_button_id is mandatory for CUSTOM_BUTTON notification
+  if (!(*message_)[app_mngr::strings::msg_params].keyExists(
+          hmi_response::custom_button_id)) {
+    LOG4CXX_ERROR(
+        logger_, "CUSTOM_BUTTON mobile notification without custom_button_id.");
+    return;
+  }
+
+  if (!app) {
+    LOG4CXX_ERROR(logger_, "Application doesn't exist.");
+    return;
+  }
+
+  uint32_t custom_btn_id = 0;
```

Comment:
@mked-luxoft What do you need intermediate initialization here for?

Reply:
@AByzhynar fixed in 704a3e5e302aba1557830edf5e6afedb18c317e3